ships/0037-build-output-timestamp.md

```diff
@@ -0,0 +1,142 @@
+<!--
+Copyright The Shipwright Contributors
+
+SPDX-License-Identifier: Apache-2.0
+-->
+
+---
+title: build-output-timestamp
+authors:
+  - "@HeavyWombat"
+reviewers:
+  - "@adambkaplan"
+  - "@qu1queee"
+  - "@SaschaSchwarze0"
+approvers:
+  - "@adambkaplan"
+  - "@qu1queee"
+creation-date: 2023-04-24
+last-updated: 2023-04-24
+status: implementable
+---
+
+# Build Output Image Timestamp
+
+## Release Signoff Checklist
+
+- [x] Enhancement is `implementable`
+- [x] Design details are appropriately documented from clear requirements
+- [x] Test plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, DA
+- [ ] User-facing documentation is created in [docs](/docs/)
+
+## Summary
+
+This _ship_ proposes an extension to the `spec.output` of a Build (or BuildRun through the build specification) to explicitly set the result image timestamp.
+
+## Motivation
+
+With Shipwright, we can help users to achieve [reproducible builds](https://reproducible-builds.org/docs/timestamps/) depending on the tools that are used in the strategy. One factor of reproducible builds is avoiding the usage of build system specific timestamps in the final image. Ideally, when using a `Dockerfile` based build, running the build twice with the exact same source input should yield the same image.
+
+### Goals
+
+- Allow a Build user to specify that a neutral timestamp is used for the creation time of the container image, i.e. unix timestamp zero.
+- Allow a Build user to specify that -- if possible -- Shipwright is suppose to come up with a reasonable deterministic timestamp of the build, e.g. the Git commit timestamp for Git source based builds similar to how GoReleaser does it.
+- Allow a Build user to specify that Shipwright is not suppose to do anything with regards to the container image creation timestamp.
+
+### Non-Goals
+
+- Modification of timestamps of the source files.
+- No post-processing of container images that used strategy managed push, i.e. Paketo Buildpacks.
+- No magic build strategy manipulation of steps other than the `image-processing` step.
+
+### User Stories
+
+#### Story 1
+
+As a Shipwright user, I want to use a `Dockerfile` based build where the resulting container image creation timestamp is set to unix timestamp zero.
+
+#### Story 2
+
+As a Shipwright user, I want to use build a container image that has the image creation timestamp that matches the one of the Git commit that was used for the build.
+
+#### Story 3
+
+As a Shipwright user, I want write a build that results in a build run that does not touch the image timestamp at all.
+
+## Proposal
+
+### API Changes
+
+We will extend the Build resource by an additional (Unix epoch) timestamp entry in `spec.output`:
```

Comment:
pls also include  on the  `BuildRunStatus` object, we need to know what will change there, .e.g.  how are we gonna surface the timestamp value to users via the BuildRun status?

Reply:
With regards to the spec, it is covered a few sentences later:
> The same fields will be made available in a BuildRun's `spec.output` and `spec.buildSpec.output`.